repos/system_upgrade/el8toel9/actors/multipathconfupdate/tests/test_multipath_conf_update_8to9.py

```diff
@@ -0,0 +1,176 @@
+    monkeypatch.setattr(multipathutil, 'write_config', mock_write_config)
+
+
+default_rhel8_conf = build_config(
+    ['default_rhel8.conf', None, True, False, False, ])
+
+all_the_things_conf = build_config(
+    ['all_the_things.conf', None, False, True, False, ])
+
+converted_the_things_conf = build_config(
+    ['converted_the_things.conf', None, True, False, True, ])
+
+idempotent_conf = build_config(
+    ['converted_the_things.conf', None, False, True, False, ])
+
+complicated_conf = build_config(
+    ['complicated.conf', '/etc/multipath/conf.d', True, True, False, ])
+
+no_foreign_conf = build_config(
+    ['no_foreign.conf', None, False, True, True, ])
+
+allow_usb_conf = build_config(
+    ['allow_usb.conf', None, False, False, True, ])
+
+no_defaults_conf = build_config(
+    ['no_defaults.conf', None, False, True, False, ])
+
+two_defaults_conf = build_config(
+    ['two_defaults.conf', None, True, False, False, ])
+
+empty_conf = build_config(
+    ['empty.conf', None, False, False, False, ])
+
+missing_dir_conf = build_config(
+    ['missing_dir.conf', 'missing', False, True, False, ])
+
+not_set_dir_conf = build_config(
+    ['not_set_dir.conf', 'conf1.d', False, True, False, ])
+
+empty1_conf = build_config(
+    ['conf1.d/empty.conf', None, False, False, False, ])
+
+nothing_important_conf = build_config(
+    ['conf1.d/nothing_important.conf', 'this_gets_ignored', False, False, False, ])
+
+set_in_dir_conf = build_config(
+    ['set_in_dir.conf', 'conf2.d', False, False, False, ])
+
+all_true_conf = build_config(
+    ['conf2.d/all_true.conf', None, True, True, True, ])
+
+empty_dir_conf = build_config(
+    ['empty_dir.conf', 'conf3.d', False, False, False, ])
+
+default_rhel8_facts = build_facts([default_rhel8_conf])
+all_the_things_facts = build_facts([all_the_things_conf])
+converted_the_things_facts = build_facts([converted_the_things_conf])
+complicated_facts = build_facts([complicated_conf])
+no_foreign_facts = build_facts([no_foreign_conf])
+allow_usb_facts = build_facts([allow_usb_conf])
+no_defaults_facts = build_facts([no_defaults_conf])
+two_defaults_facts = build_facts([two_defaults_conf])
+empty_facts = build_facts([empty_conf])
+missing_dir_facts = build_facts([missing_dir_conf])
+empty_dir_facts = build_facts([empty_dir_conf])
+not_set_dir_facts = build_facts([not_set_dir_conf, empty1_conf, nothing_important_conf])
+set_in_dir_facts = build_facts([set_in_dir_conf, all_true_conf])
+idempotent_facts = build_facts([idempotent_conf])
+
+
+def _test_facts(facts):
+    multipathconfupdate.update_configs(facts)
+    for config in facts.configs:
+        expected_data = multipathutil.read_config_orig(os.path.join(AFTER_DIR, config.pathname))
+        if config.pathname in converted_data:
+            assert converted_data[config.pathname] == expected_data
+        else:
+            assert expected_data is None
+
+
+def test_default_rhel8(setup_test):
```

Comment:
Same suggestion as above - instead of N similar tests have a mapping and just 1.

Reply:
Yep.